src/main/java/net/fabricmc/api/FakeModProvider.java

```diff
@@ -0,0 +1,13 @@
+package net.fabricmc.api;
+
+import net.fabricmc.loader.api.ModContainer;
```

Comment:
There is currently not way to distinguish between a normal and fake mod container.
First I think we should add a isFake/isProvided method.

I think we should have return a `ProvidedModContainer` which extends `ModContainer` and a `ModContainer getProvider();` so we could track which mod made the fake mod. Of course this would require a way to get what ModContainer provides an entrypoint (so we have a source of the FakeModProvider).